python/ray/serve/backend_state.py

```diff
@@ -816,6 +830,16 @@ def update(self) -> bool:
 
         transitioned_backend_tags = set()
         for backend_tag, replicas in self._replicas.items():
+            for replica in replicas.pop(states=[ReplicaState.RUNNING]):
+                if replica.check_health():
+                    replicas.add(ReplicaState.RUNNING, replica)
+                else:
+                    logger.warning(
+                        f"Replica {replica.replica_tag} of backend "
+                        f"{backend_tag} failed health check, stopping it.")
+                    replica.set_should_stop(0)
```

Comment:
replica.stop(force=True) should also work here right? I have no idea what `set_should_stop(0)` do and it's not consistent with other action here (replica.start, stop, etc)

Reply:
The state transition is `RUNNING` -> `SHOULD_STOP` -> `STOPPING`, we can't directly stop the replicas in case the controller crashes in between.